pkg/scheduler/framework/plugins/noderesources/balanced_allocation.go

```diff
@@ -37,10 +37,49 @@ type BalancedAllocation struct {
 	resourceAllocationScorer
 }
 
+var _ framework.PreScorePlugin = &BalancedAllocation{}
 var _ framework.ScorePlugin = &BalancedAllocation{}
 
 // BalancedAllocationName is the name of the plugin used in the plugin registry and configurations.
-const BalancedAllocationName = names.NodeResourcesBalancedAllocation
+const (
+	BalancedAllocationName = names.NodeResourcesBalancedAllocation
+
+	// balancedAllocationPreScoreStateKey is the key in CycleState to NodeResourcesBalancedAllocation pre-computed data for Scoring.
+	balancedAllocationPreScoreStateKey = "PreScore" + BalancedAllocationName
+)
+
+// balancedAllocationPreScoreState computed at PreScore and used at Score.
+type balancedAllocationPreScoreState struct {
+	podRequest map[v1.ResourceName]int64
+}
+
+// Clone implements the mandatory Clone interface. We don't really copy the data since
+// there is no need for that.
+func (s *balancedAllocationPreScoreState) Clone() framework.StateData {
+	return s
+}
+
+// PreScore calculates incoming pod's resource requests and writes them to the cycle state used.
+func (ba *BalancedAllocation) PreScore(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodes []*v1.Node) *framework.Status {
+	state := &balancedAllocationPreScoreState{
+		podRequest: ba.calculatePodResourceRequestMap(pod, ba.resources),
+	}
```

Comment:
As you know, now we've got the Skip in PreScore.
https://github.com/kubernetes/kubernetes/issues/115745

Can we return skip if Pod has no resource request?
Please double-check [the scorer logic ](https://github.com/kubernetes/kubernetes/blob/268c02a1c7f63d760af7c2bf614e93f95ea80e9d/pkg/scheduler/framework/plugins/noderesources/balanced_allocation.go#L136-L171)(+ somewhere else) to see if it's really OK to skip in that case.

Reply:
Even if the pod requests are zero, this plugin would still return different scores (because we consider what's already in the node).

Reply:
> Even if the pod requests are zero, this plugin would still return different scores (because we consider what's already in the node).

I think sanposhiho means resources not set, for example the test https://github.com/kubernetes/kubernetes/blob/04675428bbfc9bf7ba4c9e1abfc427b6228069d9/pkg/scheduler/framework/plugins/noderesources/least_allocated_test.go#L87-L100